examples/tool-use-vercel/src/financial-analysis.service.ts

```diff
@@ -0,0 +1,190 @@
+          parameters: z.object({
+            prompt: z.string().describe("The natural language query/prompt to extract data from the page"),
+          }),
+          execute: async ({ prompt }) => {
+            this.log.info(`Extracting data from the page with prompt: ${prompt}`);
+            if (!this.sessionId || !this.windowId) {
+              throw new Error("Session or window ID not found");
+            }
+
+            const result = await this.client.windows.pageQuery(this.sessionId, this.windowId, { prompt });
+            return {
+              data: result.data.modelResponse,
+            };
+          },
+        }),
+        // Tool to click an element on the page
+        clickButton: tool({
+          description: "Click an element on the page",
+          parameters: z.object({
+            elementDescription: z.string().describe("The description of the element to click"),
+          }),
+          execute: async ({ elementDescription }) => {
+            this.log.info(`Clicking element on the page with description: ${elementDescription}`);
+            if (!this.sessionId || !this.windowId) {
+              throw new Error("Session or window ID not found");
+            }
+
+            const result = await this.client.windows.click(this.sessionId, this.windowId, { elementDescription });
+            return {
+              data: result.data.modelResponse,
+            };
+          },
+        }),
+      },
+      maxSteps: 20, // allow up to 20 steps
+      prompt: `
+      You are a financial analyst. You are given a goal and a set of tools to use to achieve the goal.
+      You will use the tools to achieve the goal.
+      You will return the result of the goal and the steps you took to achieve it.
+      You may only load one URL at a time, and you may only click one element at a time.
+      
+      When accessing historical information:
+      1. Click on ONE time period button (e.g., "1D" or "5D")
+      2. IMMEDIATELY after clicking, extract the data for that time period
+      3. Move on to the next time period and repeat the process if necessary
+      
+      Do not click multiple buttons in sequence without extracting data between clicks.
+      Each historical data point should follow this strict pattern:
+      - Click time period button
+      - Extract data
+      - Move to next time period
+
+      Goal: ${goal}
+      `,
+    });
+
+    return {
+      text,
+      steps,
+    };
+  }
+
+  /**
+   * Performs a financial analysis.
+   * @param {string} sessionId - The ID of the session
+   * @param {string} windowId - The ID of the window
+   * @returns {Promise<string>} The result of the analysis
+   */
+  async performAnalysis(sessionId: string, windowId: string): Promise<string> {
+    this.sessionId = sessionId;
+    this.windowId = windowId;
+
+    const goal = `Extract the current price of NVDA and the growth percentage over all the time periods available. 
```

Comment:
We might want to move this to a constant at the top of the file and add a comment to edit this to change from `NVDA`. We could also add this comment to the Fern docs and not edit in this PR. Up to you.